### PR TITLE
Add backoff to debuggee registration

### PR DESCRIPTION
**Pull request by hxiong388**

### CODE REVIEW

File: google-cloud-debugger/test/google/cloud/debugger/backoff_test.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      backoff.failed
+      interval2 = backoff.interval
+
+      (interval2 > interval1).must_equal true
+    end
+  end
+
+  describe "#wait" do
+    it "doesn't sleep if " do
+
```
Comment (dazuma) [inline google-cloud-debugger/test/google/cloud/debugger/backoff_test.rb]: ?

Comment (hxiong388) [reply google-cloud-debugger/test/google/cloud/debugger/backoff_test.rb]: Oops, left that out from the refactoring.

**Comment by hxiong388** [comment]
@googlebot Is this how I find out I'm fired?

**Comment by hxiong388** [comment]
I signed it!

**Pull request merged**